ujson may caused test failed

Env:
- Python 3.7.2 [GCC 8.2.1 20181127] on linux
- bottle master branch, git sha `c8179b28d93b2875a31866c6b84a9b5b59c0c8b4`
- usjon 1.35

```
======================================================================
FAIL: test_json_serialization_error (test.test_outputfilter.TestOutputFilter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/bottle/test/test_outputfilter.py", line 86, in test_json_serialization_error
    self.assertStatus(500)
  File "/root/bottle/test/tools.py", line 139, in assertStatus
    self.assertEqual(code, self.urlopen(route, **kargs)['code'])
AssertionError: 500 != 200

======================================================================
FAIL: test_json_plugin_catches_httpresponse (test.test_plugins.TestPluginManagement)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/bottle/test/test_plugins.py", line 163, in test_json_plugin_catches_httpresponse
    self.assertBody(b'{"test": "ko"}', '/return')
  File "/root/bottle/test/tools.py", line 142, in assertBody
    self.assertEqual(tob(body), self.urlopen(route, **kargs)['body'])
AssertionError: b'{"test": "ko"}' != b'{"test":"ko"}'

----------------------------------------------------------------------
Ran 341 tests in 5.847s

FAILED (failures=2)
```
The first one is because `usjon` support serialize a `set`

```Python
>>> ujson.dumps(set())
'{"a":[]}'
>>> ujson.dumps({'a': object()})  # even an object
'{"a":{}}'
```

The second one is because

```Python
>>> d = {"test": "ko"}
>>> json.dumps(d)
'{"test": "ko"}'
>>> ujson.dumps(d)
'{"test":"ko"}'  # no space after colon
```

--- a/test/test_outputfilter.py
+++ b/test/test_outputfilter.py
@@ -6,6 +6,13 @@ import bottle
 from bottle import tob, touni
 from .tools import ServerTestBase, tobs, warn
 
+USING_UJSON = True
+
+try:
+    import ujson
+except ImportError:
+    USING_UJSON = False
+
 class TestOutputFilter(ServerTestBase):
     ''' Tests for WSGI functionality, routing and output casting (decorators) '''
 
@@ -76,6 +83,7 @@ class TestOutputFilter(ServerTestBase):
         except ImportError:
             warn("Skipping JSON tests.")
 
+    @unittest.skipIf(USING_UJSON, 'ujson do not throw exception in serialize')
     def test_json_serialization_error(self):
         """
         Verify that 500 errors serializing dictionaries don't return
@@ -87,6 +95,7 @@ class TestOutputFilter(ServerTestBase):
             self.assertHeader('Content-Type','text/html; charset=UTF-8')
         except ImportError:
             warn("Skipping JSON tests.")
+
 
     def test_json_HTTPResponse(self):
         self.app.route('/')(lambda: bottle.HTTPResponse({'a': 1}, 500))

--- a/test/test_outputfilter.py
+++ b/test/test_outputfilter.py
@@ -77,11 +77,9 @@ class TestOutputFilter(ServerTestBase):
 
     def test_json(self):
         self.app.route('/')(lambda: {'a': 1})
-        try:
-            self.assertBody(bottle.json_dumps({'a': 1}))
-            self.assertHeader('Content-Type','application/json')
-        except ImportError:
-            warn("Skipping JSON tests.")
+
+        self.assertBody(bottle.json_dumps({'a': 1}))
+        self.assertHeader('Content-Type','application/json')
 
     @unittest.skipIf(USING_UJSON, 'ujson do not throw exception in serialize')
     def test_json_serialization_error(self):
@@ -90,29 +88,23 @@ class TestOutputFilter(ServerTestBase):
         content-type application/json
         """
         self.app.route('/')(lambda: {'a': set()})
-        try:
-            self.assertStatus(500)
-            self.assertHeader('Content-Type','text/html; charset=UTF-8')
-        except ImportError:
-            warn("Skipping JSON tests.")
+
+        self.assertStatus(500)
+        self.assertHeader('Content-Type','text/html; charset=UTF-8')
 
 
     def test_json_HTTPResponse(self):
         self.app.route('/')(lambda: bottle.HTTPResponse({'a': 1}, 500))
-        try:
-            self.assertBody(bottle.json_dumps({'a': 1}))
-            self.assertHeader('Content-Type','application/json')
-        except ImportError:
-            warn("Skipping JSON tests.")
+
+        self.assertBody(bottle.json_dumps({'a': 1}))
+        self.assertHeader('Content-Type','application/json')
 
     def test_json_HTTPError(self):
         self.app.error(400)(lambda e: e.body)
         self.app.route('/')(lambda: bottle.HTTPError(400, {'a': 1}))
-        try:
-            self.assertBody(bottle.json_dumps({'a': 1}))
-            self.assertHeader('Content-Type','application/json')
-        except ImportError:
-            warn("Skipping JSON tests.")
+
+        self.assertBody(bottle.json_dumps({'a': 1}))
+        self.assertHeader('Content-Type','application/json')
 
     def test_generator_callback(self):
         @self.app.route('/')

--- a/test/test_plugins.py
+++ b/test/test_plugins.py
@@ -2,7 +2,7 @@
 import unittest
 from . import tools
 
-from bottle import HTTPResponse, HTTPError
+from bottle import HTTPResponse, HTTPError, json_dumps
 
 
 class MyPlugin(object):
@@ -160,8 +160,8 @@ class TestPluginManagement(tools.ServerTestBase):
         def _():
             raise HTTPResponse({'test': 'ko2'}, 402)
 
-        self.assertBody(b'{"test": "ko"}', '/return')
-        self.assertBody(b'{"test": "ko2"}', '/raise')
+        self.assertBody(json_dumps({'test': 'ko'}), '/return')
+        self.assertBody(json_dumps({'test': 'ko2'}), '/raise')
 
 
 class TestPluginAPI(tools.ServerTestBase):

